smart/differential_rotation.py

```diff
@@ -0,0 +1,33 @@
+import numpy as np
+from matplotlib import colors
+
+from sunpy.coordinates import propagate_with_solar_surface
+from sunpy.map import all_coordinates_from_map, coordinate_is_on_solar_disk
+
+
+def diff_rotation(im_map, delta_im_map):
+    """
+    Performing differential rotation on a map in order to to correct for feature
+    motions due to solar rotation.
+
+    Parameters
+    ----------
+    im_map : sunpy.map.Map
```

Comment:
Make sure to import map at the top then

```suggestion
    im_map : `Map`
```